### PR TITLE
:books: add token macro example

### LINKED ISSUE #31: delete builds if a certain parameter is true or matches a value
I have a parameterized job for running running taurus tests. It is triggered by other jobs that pass in paramaters specific to the test to be run. E.g. a smoke test, a large load test, a small load test, etc -- and thus functions as a test coordinator and handles locking, reporting, etc. etc. in a single place. 

We have a scheduled job that triggers our test runner job telling it to run a smoke test every 5 minutes. 

As such, we are getting **a lot** of build history.  these scheduled smoke tests builds **don't need to be stored**. but we'd like to store the manually run builds. 

A boolean parameter is used to differentiate between the manual and scheduled tests, e.g.

```
parameters {
    booleanParam(name: 'ENABLE_HISTORY', defaultValue: true, description: 'Check to preserve build history.')
    ...
```

and we pass `ENABLE_HISTORY` as false on scheduled builds.

Is it possible to use this plugin to match this condition? I see the Token macro implementation but am confused as to how it works / don't see any examples.

Many thanks!

### PR DESCRIPTION
**Pull request by briceburg**
Closes #31

### CODE REVIEW

**Comment by damianszczepanik** [comment]
How nice!

File: README.md [hunk 1]
```diff
@@ -52,6 +55,54 @@ pipeline {
...
+
+```groovy
+pipeline {
+    agent any
+
+    options {
+        buildDiscarder(BuildHistoryManager([
+            [
+                conditions: [
+                    TokenMacro(template: '"${ENABLE_HISTORY}"', value: '"false"')
```
Comment (damianszczepanik) [inline README.md:72]: do we need double quotes for `false`?

File: README.md [hunk 2]
```diff
@@ -52,6 +55,54 @@ pipeline {
 }
 ```
 
+#### remove builds based on a parameter
+
+Following configuration has two rules. The first uses the token macro condition to test
+the value of a parameter. The second rule will preserve the last 24 builds. In other words
```
Comment (damianszczepanik) [inline README.md:61]: ...last 24 builds for which the parameter was set to false

File: README.md (hunk 1 above)
Comment (briceburg) [reply README.md:72]: I did find this to be the case

**Comment by briceburg** [comment]
great suggestions! I'll followup w/ a PR that addresses these.

**Pull request merged**